generators/entity/index.js

```diff
@@ -61,7 +61,12 @@ module.exports = EntityGenerator.extend({
             type: Boolean,
             defaults: false
         });
+        this.option('table-name', {
+           desc: 'specify table name that will be used by the entity',
+            type: String
+        });
         this.regenerate = this.options['regenerate'];
+        this.tableName = this.options['table-name'];
```

Comment:
Use variable entityTableName instead of tableName
And underscore and lowercase the passed value here